connectionbuddy/src/main/java/com/zplesac/connectionbuddy/ConnectionBuddy.java

```diff
@@ -117,7 +130,11 @@ public ConnectionBuddyConfiguration getConfiguration() {
      * @param object   Object which is registered to network change receiver.
      * @param listener Callback listener.
      */
-    public void registerForConnectivityEvents(Object object, ConnectivityChangeListener listener) {
+    public void registerForConnectivityEvents(@NonNull Object object, @NonNull ConnectivityChangeListener listener) {
+        if (configuration == null) {
```

Comment:
Let's extract this to a separate method check as we are using it trough on a couple places inside this class.